weaviate/collection/grpc.py

```diff
@@ -553,20 +556,33 @@ def __result_to_object(
         metadata = self._extract_metadata_for_object(res.additional_properties)
         return _Object[Properties](properties=properties, metadata=metadata)
 
+    def __determine_generic(
+        self, type_: Union[PROPERTIES, Type[Properties], None]
+    ) -> Tuple[Optional[PROPERTIES], Union[Dict[str, Any], Properties]]:
+        if isinstance(type_, list) or isinstance(type_, str) or type_ is None:
+            ret_properties = cast(Optional[PROPERTIES], type_)
+            ret_type = Dict[str, Any]
+        else:
+            assert get_origin(type_) is not dict
+            type_ = cast(Type[Properties], type_)
+            ret_properties = _extract_properties_from_data_model(type_)
+            ret_type = type_
+        return ret_properties, ret_type
+
     def get_flat(
         self,
+        return_properties: Union[PROPERTIES, Type[Properties], None] = None,
+        return_metadata: Optional[MetadataQuery] = None,
```

Comment:
This was done in case they ever stop being optional, there was an experimental use-case I was messing with that required this 

Perhaps, in hindsight, this will always be optional as a fundamental requirement so will change back